tests/components/orderentry/OrdersTable.test.jsx

```diff
@@ -123,3 +181,33 @@ describe('Orders component test-suite', () => {
     expect(props.dispatch).toBeCalled();
   });
 });
+
+describe('getUUID() method', () => {
+  it('should call getUUID()', () => {
+    const renderedComponent = getComponent().instance();
+    sinon.spy(renderedComponent, 'getUUID');
+    renderedComponent.getUUID(items, itemName);
+    expect(renderedComponent.getUUID.calledOnce).toEqual(true);
+  });
+});
+
+describe('setDiscontinuedDrugOrder method', () => {
+  it('should call setDiscontinuedDrugOrder', async (done) => {
+    const renderedComponent = getComponent().instance();
+    sinon.spy(renderedComponent, 'setDiscontinuedDrugOrder');
+    renderedComponent.setDiscontinuedDrugOrder(mockOrder);
+    expect(renderedComponent.setDiscontinuedDrugOrder.calledOnce).toEqual(true);
+    done();
+  });
+});
+
+describe('discontinueDrugOrder method', () => {
+  it('should call discontinueDrugOrder', async (done) => {
+    const renderedComponent = getComponent().instance();
+    sinon.spy(renderedComponent, 'discontinueDrugOrder');
+    renderedComponent.discontinueDrugOrder(mockOrder, 22);
+    expect(renderedComponent.discontinueDrugOrder.calledOnce).toEqual(true);
+    expect(props.dispatch).toBeCalled();
+    done();
```

Comment:
for these test-suites, you basically just called the method and assert that it was called, the test-suites doesn't describe or show the behavior of the component with respect to these actions that are happening, remember, this is a test for a component(UI) and not just a javascript class.

This scenario is a good situation to use `mount` so as to be able to simulate clicks before asserting a function call.

Reply:
I agree that you can't simulate clicks since you are using shallow, that's why I proposed you use mount to simulate events instead of calling the method directly. however, the real issue is that the tests are not pragmatic.

 You are basically calling these methods directly and asserting that they were called, of course, it would be called because you just did that directly  😄. a better approach would have been to assert based on the effect of calling that method on the component, for instance, the `handleDiscard` method, when called, what happens in the component?

`is a toast shown?` `is the discarded element removed from the view?`

your assertions should be based off that, and not just direct method calls.

Reply:
So I believe we also use this convention here in this code base and I don't think making my assertion to simulate the whole clicking behaviour would change anything or affect the coverage, and last I checked it was unit testing and not integration tests, so I'll be leaving this as it is too